packages/flutter_tools/lib/src/build_system/targets/ios.dart

```diff
@@ -365,6 +367,125 @@ class DebugUnpackIOS extends UnpackIOS {
   BuildMode get buildMode => BuildMode.debug;
 }
 
+// TODO(gaaclarke): Remove this after a reasonable amount of time where the
+// UISceneDelegate migration being on stable. This incurs a minor build time
+// cost.
+Future<void> _checkForLaunchRootViewControllerAccessDeprecation(
+  Logger logger,
+  File file,
+  Pattern usage,
+  Pattern terminator,
+) async {
+  final Stream<String> lines = file
+      .openRead()
+      .transform(utf8.decoder)
+      .transform(const LineSplitter());
+
+  bool inDidFinishLaunchingWithOptions = false;
+  int lineNumber = 0;
+  await for (final String line in lines) {
+    lineNumber += 1;
+    if (!inDidFinishLaunchingWithOptions) {
+      if (line.contains('didFinishLaunchingWithOptions')) {
+        inDidFinishLaunchingWithOptions = true;
+      }
+    } else {
+      if (line.startsWith(terminator)) {
+        inDidFinishLaunchingWithOptions = false;
```

Comment:
There's no reason to search for `didFinishLaunchingWithOptions` again, is there?

```suggestion
        break;
```